app/models/miq_widget_set.rb

```diff
@@ -27,10 +28,15 @@ def self.destroy_user_versions
   end
 
   def self.where_unique_on(name, user = nil)
+    # user is nil for dashboards set for group
     userid = user.try(:userid)
     group_id = user.try(:current_group_id)
     # a unique record is defined by name, group_id and userid
-    where(:name => name, :group_id => group_id, :userid => userid)
+    if userid.present?
+      where(:name => name, :group_id => group_id, :userid => userid)
```

Comment:
You can also basically remove the group_id variable now, and just use `user.current_group_id`, since the group_id isn;t needed elsewhere, and you don't need the try because you've checked userid is present.

Reply:
👍